src/diffusers/utils/testing_utils.py

```diff
@@ -132,6 +132,33 @@ def numpy_cosine_similarity_distance(a, b):
     return distance
 
 
+def check_if_dicts_are_equal(dict1, dict2):
+    for key, value in dict1.items():
+        if isinstance(value, set):
+            dict1[key] = sorted(value)
+    for key, value in dict2.items():
+        if isinstance(value, set):
+            dict2[key] = sorted(value)
+
+    for key in dict1:
+        if key not in dict2:
+            raise ValueError(
+                f"Key '{key}' is missing in the second dictionary. Its value in the first dictionary is {dict1[key]}."
+            )
+        if dict1[key] != dict2[key]:
+            raise ValueError(
+                f"Difference found at key '{key}': first dictionary has {dict1[key]}, second dictionary has {dict2[key]}."
+            )
+
+    for key in dict2:
+        if key not in dict1:
+            raise ValueError(
+                f"Key '{key}' is missing in the first dictionary. Its value in the second dictionary is {dict2[key]}."
+            )
```

Comment:
You could create sets of the keys and check the `symmetric_difference`.

Reply:
Going for the other suggestion on returning False as the current way of checking keys without `symmetric_difference` is more explicit to me and more readable.

Reply:
To me, the advantage of `symmetric_difference` is that it immediately returns all differing keys, whereas the code as is first checks for missing keys in `dict2`, then when the user fixes it, there can be another error with keys missing in `dict1`. But I agree that some readers might not know about some of these set methods.